src/app/listener.ts

```diff
@@ -77,22 +77,38 @@ export class Listener {
     this.container.clientService.on('interactionCreate', async (interaction) => {
       // Check if this is an autocomplete interaction and handle it appropriately.
       if (interaction.isAutocomplete()) {
-        await slashCommands
-          .get(interaction.commandName)
-          ?.autocomplete?.({ interaction, container: this.container });
+        const command = commands.get(interaction.commandName);
+
+        if (!command?.type || command.type === 'CHAT_INPUT') {
+          await command?.autocomplete?.({ interaction, container: this.container });
+        }
 
         return;
       }
 
       // If it's not a command, we don't care.
-      if (!interaction.isCommand()) {
+      if (!interaction.isApplicationCommand()) {
+        return;
+      }
+
+      const command = commands.get(interaction.commandName);
+
+      // For future proofing against any new command types we verify the command type
+      // being run is intentional.
+      if (interaction.isCommand() && (!command?.type || command?.type === 'CHAT_INPUT')) {
+        command?.execute({ interaction, container: this.container });
         return;
       }
 
-      // We only need the slash command handler.
-      await slashCommands
-        .get(interaction.commandName)
-        ?.execute({ interaction, container: this.container });
+      if (interaction.isUserContextMenu() && command?.type === 'USER') {
+        command?.execute({ interaction, container: this.container });
+        return;
+      }
+
+      if (interaction.isMessageContextMenu() && command?.type === 'MESSAGE') {
+        command?.execute({ interaction, container: this.container });
+        return;
+      }
```

Comment:
Do these need to be spread out? like whats the use case where we care where the command came from